src/linear_algebra/char_poly/coeff.lean

```diff
@@ -156,6 +156,27 @@ begin
 end
 
 variables {p : ℕ} [fact p.prime]
+#where
+-- the following is a nasty case bash that I (DT) abstracted from the lemma below
+-- a pre-existing comment asked "maybe it can be proven more... algebraically?"
+lemma mat_poly_equiv_eq_X_pow_sub_C {K : Type*} (p k : ℕ) [field K] (M : matrix n n K) :
+  mat_poly_equiv
+      (((expand K (p ^ k) : polynomial K →+* polynomial K).map_matrix) (char_matrix (M ^ p ^ k))) =
+    X ^ p ^ k - C (M ^ p ^ k) :=
```

Comment:
Can you generalize `p ^ k` to just `pk`?

Reply:
Might be: for the moment, I simply wanted to speed up the lemma that gave a timeout and to check what else was broken.  But you are right: this lemma needs some love...

Reply:
My hope would be that you don't actually have to change the proof to use `pk` instead.

Reply:
Ok, a new file broke: I may not have time to fix that file today, but this suggestion is in.